mlflow/utils/nfs_on_spark.py

```diff
@@ -25,7 +29,18 @@ def get_nfs_cache_root_dir():
             == "true"
         )
         if nfs_enabled:
-            return "/local_disk0/.ephemeral_nfs"
+            nfs_root_dir = "/local_disk0/.ephemeral_nfs"
+            # Test whether the NFS directory is writable.
+            test_path = os.path.join(nfs_root_dir, uuid.uuid4().hex)
+            try:
+                os.makedirs(test_path)
+                return nfs_root_dir
+            except Exception:
+                # For databricks cluster enabled Table ACL, we have no permission to access NFS
+                # directory, in this case, return None representing NFS is not available.
+                return None
+            finally:
+                shutil.rmtree(test_path, ignore_errors=True)
```

Comment:
Does this line fail when `test_path` doesn't exist?

Reply:
Good catch! Fortunately, it swallows:

```
In [11]: shutil.rmtree("/tmp/asfsadfsdf", ignore_errors=True)

In [12]: shutil.rmtree("/tmp/asfsadfsdf")
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-12-8bc109b15d91> in <module>
----> 1 shutil.rmtree("/tmp/asfsadfsdf")

~/opt/anaconda3/lib/python3.8/shutil.py in rmtree(path, ignore_errors, onerror)
    704             orig_st = os.lstat(path)
    705         except Exception:
--> 706             onerror(os.lstat, path, sys.exc_info())
    707             return
    708         try:

~/opt/anaconda3/lib/python3.8/shutil.py in rmtree(path, ignore_errors, onerror)
    702         # lstat()/open()/fstat() trick.
    703         try:
--> 704             orig_st = os.lstat(path)
    705         except Exception:
    706             onerror(os.lstat, path, sys.exc_info())

FileNotFoundError: [Errno 2] No such file or directory: '/tmp/asfsadfsdf'
```

Reply:
Confirmed it doesn't